### DISCUSSION TITLE
How to self host?

### DISCUSSION THREAD

**Discussion by ottobaer** [MicroPad/Web#458] [Q&A] [answered]
Hi, 

I followed the documentation on how to build it with bun. 
Now how to I start the production server so I can put some reverse proxy before it to host it for myself? 

Thank You,

**Comment by NickGeek** [comment, marked as answer]
Hi @ottobaer. MicroPad runs client-side so you do not need any production server beyond a simple server that serves files (i.e. nginx). So no reverse proxy is needed (unless you want to reverse proxy to another nginx, apache, etc. server).

I'd put the built files in a directory that you are serving with your web server and it should all just work. By default MicroPad expects to run at the root of a website (i.e. with its index.html at /) so you may want to put it on a sub-domain.

MicroPad has two optional non-local dependencies, one is a list of currencies (https://getmicropad.com/cors-proxy/moolah.xml) and the other is MicroSync (https://getmicropad.com/sync). You can change the base-url for MicroPad to your URL in https://github.com/MicroPad/MicroPad-Core/blob/b5fe4bc68d708ec5fcd91f3e469c6a5d6438b892/app/src/app/types/index.tsx#L65. However, the server for MicroSync is not freely available so you may just want to either leave it pointing to getmicropad.com (in `services/DifferenceEngine.ts`) or point it to some URL that does not exist and allow it to fail.

You can still easily sync notebooks without MicroSync by saving your notebooks to a synced folder (i.e. dropbox, onedrive, iCloud Drive) with Ctrl+S/Cmd+S and Ctrl+O/Cmd+O to open a notebook (Ctrl+Shift+S/Cmd+Shift+S to save all of your notebooks at once to a single archive, these can be opened the same way as a single notebook).

**Comment by ottobaer** [comment]
Thanks a lot, just got it to run.
